### PR TITLE
refactor: Use early return pattern to avoid nested conditions

### PR DESCRIPTION
**Pull request by jongwooo**
Signed-off-by: jongwooo <jongwooo.han@gmail.com>
Issue #297 

## Description

Return early is the way of writing functions or methods so that the expected positive result is returned at the end of the function and the rest of the code terminates the execution (by returning or throwing an exception) when conditions are not met.

See [actions/cache#1012](https://github.com/actions/cache/issues/1012), [actions/setup-node#639](https://github.com/actions/setup-node/pull/639)

In `cache-utils.ts`:

**AS-IS**

```typescript
export function isCacheFeatureAvailable(): boolean {
  if (!cache.isFeatureAvailable()) {
    if (isGhes()) {
      throw new Error(
        'Cache action is only supported on GHES version >= 3.5...'
      );
    } else {
      core.warning(
        'The runner was not able to contact...'
      );
    }

    return false;
  }

  return true;
}
```

**TO-BE**

```typescript
export function isCacheFeatureAvailable(): boolean {
  if (cache.isFeatureAvailable()) {
    return true;
  }

  if (isGhes()) {
    throw new Error(
      'Cache action is only supported on GHES version >= 3.5...'
    );
  }

  core.warning(
    'The runner was not able to contact...'
  );
  return false;
}
```

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by panticmilos** [comment]
hi @jongwooo again :D thank you for the PR. We will take a look at it

File: src/cache-utils.ts [hunk 1]
```diff
@@ -57,19 +57,18 @@ export function isGhes(): boolean {
...
-        'The runner was not able to contact the cache service. Caching will be skipped'
-      );
-    }
+  if (cache.isFeatureAvailable()) {
+    return true;
+  }
 
-    return false;
+  if (isGhes()) {
+    throw new Error(
```
[the remark this answers is not in the document]
Comment (jongwooo) [reply src/cache-utils.ts]: @e-korolevskii Thank you for your review! I also updated the testcode.

**Review by e-korolevskii — APPROVED** [review]

**Review by marko-zivic-93 — APPROVED** [review]

File: __tests__/cache-utils.test.ts [hunk 2]
```diff
@@ -174,6 +174,7 @@ describe('isCacheFeatureAvailable', () => {
       'Cache action is only supported on GHES version >= 3.5. If you are on version >=3.5 Please check with GHES admin if Actions cache service is enabled or not.';
 
     //Act + Assert
-    expect(() => cacheUtils.isCacheFeatureAvailable()).toThrow(errorMessage);
+    expect(cacheUtils.isCacheFeatureAvailable()).toBeFalsy();
+    expect(warningSpy).toHaveBeenCalledWith(errorMessage);
```
Comment (MaksimZhukov) [inline __tests__/cache-utils.test.ts:178]: Please rename the `errorMessage` variable to `warningMessage`. 
I think we also need to rename this test case to `should warn when cache feature is unavailable and GHES is used`

Comment (jongwooo) [reply __tests__/cache-utils.test.ts]: @MaksimZhukov Thank you for your review. I pushed a commit.

**Review by MaksimZhukov — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**